internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicMetadataFetcher.java

```diff
@@ -150,22 +155,25 @@ class TopicMetadataFetcher implements Closeable {
   }
 
   // acquire the consumer from the pool
-  private PubSubConsumerAdapter acquireConsumer() {
+  PubSubConsumerAdapter acquireConsumer() {
     try {
-      return pubSubConsumerPool.take();
+      consumerWaitListSize.incrementAndGet();
+      long startTime = System.nanoTime();
+      PubSubConsumerAdapter consumerAdapter = pubSubConsumerPool.take();
+      stats.recordLatency(CONSUMER_ACQUISITION_WAIT_TIME, startTime);
```

Comment:
Shall we make recordLatency thread safe?

Reply:
I believe metric code paths are already thread safe, e.g., https://github.com/tehuti-io/tehuti/blob/a50499f2be2d8fffbf0e653cada28b48a5262b19/src/main/java/io/tehuti/metrics/Sensor.java#L104

Reply:
I think was worried about `TopicManagerStats.sensorsByTypes` which is not a threadsafe data structure, but looks like we only create the contents once in the constructor and no updates to it. So, that sounds good.

One more question, who blocked `releaseConsumer` in the first place, was it `acquireConsumer`? If I understand, we should always have capacity for `releaseConsumer` to complete, right?

Reply:
That is correct, and Tehuti lib is thread safe.